### DISCUSSION TITLE
How to serialize Interpreter class in machine context?

### DISCUSSION THREAD

**Discussion by davydof** [statelyai/xstate#2189] [Q&A]
I have a root machine with spawned machine in the context

I need to store the root machine state with the context
How I can serialize the Interpreter from root machine context?

@davidkpiano

**Comment by davidkpiano** [comment]
What do you mean by serializing the interpreter? Can you clarify?

**Reply by Andarist** [reply]
@davidkpiano I think this is a question about being able to serialize/persist the "system" (with all children recursively). So basically what you have started to work on here: https://github.com/davidkpiano/xstate/pull/2197

**Reply by davydof** [reply]
@davidkpiano 
I want to store root machine and all children machines with each state and context
https://xstate.js.org/viz/?gist=5a74dc7108b9a9a5e1c4b5a91a4c3cd0

and after store that data I want to restore root machine with all children machines with state and context

**Reply by davidkpiano** [reply]
This will be possible after a couple PRs, the first one being #2197

**Reply by davydof** [reply]
hello, friendly ping :)
any updates about serialization?
